app/views/categories/_categories.html.haml

```diff
@@ -0,0 +1,18 @@
+%ul.category-summary
+  - categories.each do |category|
+    - if category.has_children?
+      - count = category.children.inject(0) do |memo, child|
```

Comment:
https://github.com/rails/rails/blob/a678c47f6d31977e2a91cc198f115c803f0fc8e1/activesupport/lib/active_support/core_ext/enumerable.rb#L20

`sum` accepts block, but calls `map` and `inject`.

In my case, calls `inject` only and loop children at once.

In tricknotes case,

```ruby
category.children.map {|child| child.posts.published.count }.sum
```

This invokes `map` and `inject` (in `sum`), loop children and array (has same size with children).
loop twice.

In mtsmfm case,

```ruby
count = category.children.sum {|child| child.posts.published.count }
```

This invokes `map` in `sum` and call `sum` recursively and call `inject`.
After all, loop twice.